Common subdirectories: scmail-1.3/doc and scmail-1.3.modify/doc
Common subdirectories: scmail-1.3/dot.scmail and scmail-1.3.modify/dot.scmail
diff -u scmail-1.3/scbayes.in scmail-1.3.modify/scbayes.in
--- scmail-1.3/scbayes.in	2004-06-08 00:08:50.000000000 +0900
+++ scmail-1.3.modify/scbayes.in	2011-01-13 22:04:07.000000000 +0900
@@ -131,12 +131,12 @@
      (sys-rename (temporary-digest-file) (digest-file)))))
 
 (define (lock)
-  (unless (eq? (create-directory* (lock-file)) #t)
-     (scmail-wformat "~a is now being updated" (table-file))
-     (scmail-wformat "or perhaps ~a is staled." (lock-file))
-     (scmail-eformat "(Please remove the lock file if it is staled.)"))
-  )
-
+  (guard (exc ((<system-error> exc)
+               (begin (scmail-wformat "~a is now being updated" (table-file))
+                      (scmail-wformat "or perhaps ~a is staled." (lock-file))
+                      (scmail-eformat "(Please remove the lock file if it is staled.)"))))
+    (sys-mkdir (lock-file) #o0770)))
+  
 (define force-learn? (make-parameter #f))
 
 (define slow? (make-parameter #f))
Common subdirectories: scmail-1.3/scmail and scmail-1.3.modify/scmail
Common subdirectories: scmail-1.3/tests and scmail-1.3.modify/tests
